go/vt/mysqlctl/mysqld.go

```diff
@@ -71,53 +69,49 @@ type Mysqld struct {
 
 // NewMysqld creates a Mysqld object based on the provided configuration
 // and connection parameters.
-func NewMysqld(config *Mycnf, dba, allprivs, app, repl *sqldb.ConnParams, enablePublishStats bool) *Mysqld {
-	noParams := sqldb.ConnParams{}
-	if *dba == noParams {
-		dba.UnixSocket = config.SocketFile
-	}
-
-	// create and open the connection pool for dba access
-	dbaMysqlStatsName := ""
-	dbaPoolName := ""
-	if enablePublishStats {
-		dbaMysqlStatsName = "MysqlDba"
-		dbaPoolName = "DbaConnPool"
-	}
-	dbaMysqlStats := stats.NewTimings(dbaMysqlStatsName)
-	dbaPool := dbconnpool.NewConnectionPool(dbaPoolName, *dbaPoolSize, *dbaIdleTimeout)
-	dbaPool.Open(dbconnpool.DBConnectionCreator(dba, dbaMysqlStats))
-
-	// create and open the connection pool for allprivs access
-	allprivsMysqlStatsName := ""
-	if enablePublishStats {
-		allprivsMysqlStatsName = "MysqlAllPrivs"
-	}
-	allprivsMysqlStats := stats.NewTimings(allprivsMysqlStatsName)
-
-	// create and open the connection pool for app access
-	appMysqlStatsName := ""
-	appPoolName := ""
-	if enablePublishStats {
-		appMysqlStatsName = "MysqlApp"
-		appPoolName = "AppConnPool"
-	}
-	appMysqlStats := stats.NewTimings(appMysqlStatsName)
-	appPool := dbconnpool.NewConnectionPool(appPoolName, *appPoolSize, *appIdleTimeout)
-	appPool.Open(dbconnpool.DBConnectionCreator(app, appMysqlStats))
-
-	return &Mysqld{
-		config:             config,
-		dba:                dba,
-		allprivs:           allprivs,
-		dbApp:              app,
-		dbaPool:            dbaPool,
-		appPool:            appPool,
-		replParams:         repl,
-		dbaMysqlStats:      dbaMysqlStats,
-		allprivsMysqlStats: allprivsMysqlStats,
-		tabletDir:          path.Dir(config.DataDir),
+func NewMysqld(config *Mycnf, dbcfgs *dbconfigs.DBConfigs, enablePublishStats bool, flags dbconfigs.DBConfigFlag) *Mysqld {
```

Comment:
Maybe put flags after dbcfgs since they are related to each other? And also it should be dbconfigFlags to be more understandable, right?